### ISSUE TITLE
Use deepsleep between measurements

### ISSUE THREAD

**Issue by IU5HKU** [richonguzman/LoRa_APRS_Wx_Station#2]
Hi Ricardo, i found this piece of code useful especially when you don't wanna setup an entire digi or igate, and yes, this isn't an issue but another proposal. Why don't add the deepsleep() function to maximize battery duration? This kind of stations are often battery operated and solar charged, the deepsleep() could help a lot. Other than this, you don't need to be active between two consecutive readings, if it doesn't rain....in this case you need to count the "click-clack" of the rain meter, but you can always link it to an interrupt routine who counts the impulse over a dedicated "wakeup" pin, it can be done with esp32 if i remember correctly...
Thank you for your time

**Comment by richonguzman** [comment]
Hi,

i was thinking about the same thing...  I will check how to do the interrupt for the rain 

with this I think I can make it sleep without any problem.

give me a time to do this as this week I'm full on making the tracker web installer ready

**Comment by IU5HKU** [comment]
Sure, there's no hurry at all. For what i remember the solution is quite straightforward, you connect the pulse of the rain meter to a pin and after you tell to esp32 to wakeup from deep sleep when this pin is activated (status chage 0 or 1, can be selected). After that, when you wakeup, you must check the reason (there is a proper function to check why we went waked up from deepsleep and from what source) and in case the reason is "external pin trigger" then you can increment the "rain meter" variable.
This is the reference: https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/system/sleep_modes.html
I think the ext0 wakeup is the most suitable for this task. There are other interesting source for the wakeup, as you can read, but mostly not compatible with the "deepest" deepsleep you can do.

**Comment by richonguzman** [comment]
yes I see there is not much difficulty , I even have something like this running on a secret project to be shared soon

sleeping Digirepeater that wakes up when Beacon must be sended or when a message is received and should be repeated and then sleeps but LoRa Modules stays listening. So it could be easy be achieved in WX station

**Comment by IU5HKU** [comment]
Great, i know this feature of the SX12xx family, you can use a digital output of the lora module when a packet is received, to wakeup the esp32 from deepsleep. Really useful, i've just a self made digi (Wemos D1 mini ESP32 and SX1278) who needs this feature, it consumes a lot to be fully battery powered all time long... :-) The worst part of this hardware is the AM1117 ldo, that is really inefficient as i've already seen in other projects i did in the past...anyway, looking forward for your code. Invaluable work, my compliments.
